fix(a11y): show bottom nav button labels

shuuji3 opened the pull request:
fix #3119

This PR includes
- showing labels below the bottom nav button icon
- a new option to hide them on the interface setting page (just below the nav button customization section)

**With labels**

![Screenshot from 2025-01-01 23-46-09](https://github.com/user-attachments/assets/edb96085-9f43-40e3-ba12-b2e1ba4c3bc1)

**Without labels (current state)**

![Screenshot from 2025-01-01 23-46-19](https://github.com/user-attachments/assets/be7d05ba-5a52-4722-aa03-4244d9a568a6)

About the default setting, this change may surprise the existing users a bit if they like the current simpler interface. On the other hand, I also understand showing labels by default can provide better accessibility.

shuuji3 committed:
fix(a11y): show bottom nav button labels

Dehelssey commented:
On deploy preview the checkbox for hiding labels does nothing 👀 (i run /e/OS (Android) with Vivaldi)

shuuji3 commented:
@Dehelssey Unlike other settings, it seems like reloading is required to change to be effective.

shuuji3 committed:
fix: make nav button label setting reactive

shuuji3 commented:
OK, I figured out how to make them reactive. Now checking should be effective immediately.

userquin commented:
uhmmm, why is `getPreferences` not reactive?

userquin commented:
I guess it will be better to move the check in the interface page to the top, between **Bottom Nvigation** and the `Choose...` paragraph:

![imagen](https://github.com/user-attachments/assets/7640fdd7-a23b-4a27-947a-db1394792417)

userquin commented:
I guess you should also change the button layout in the interface page.

shuuji3 commented:
OK, I'll add labels in that preview too, and move the option below **Bottom Navigation** header.

> uhmmm, why is `getPreferences` not reactive?

I'm not sure yet 😅

shuuji3 committed:
refactor: fix reactivity and make it composable

shuuji3 committed:
chore: move "hide bottom labels" checkbox position

shuuji3 committed:
chore: show button label in preview area

shuuji3 commented:
- moved the checkbox
- show labels in the preview area
![Screenshot of setting page without labels in preview](https://github.com/user-attachments/assets/1d413832-ae0e-4225-8960-f2ad864214fb)
![Screenshot of setting page with labels in preview](https://github.com/user-attachments/assets/0334c2cf-4101-455d-b406-707e0e1933ba)

Also, fixed the reactivity issue with `compute()` and made it composable. (Thanks @userquin for the help!)

Other than that, I feel like we should add a small space between icon and label. I'll update that tomorrow.

userquin commented:
Don't merge yet, the preview buttons not being translated, I'm going to fix it.

userquin commented:
https://m3.material.io/components/navigation-bar/guidelines

![imagen](https://github.com/user-attachments/assets/85173f85-a843-4041-8fac-e07fc047a4de)

userquin committed:
chore: cleanup imports

userquin committed:
chore: add spanish translation

Dehelssey commented:
@userquin indeed, ellipsis is not advised. If that label can't be replaced by something shorter I'm not sure how to deal with this. French isn't as effective as english but we don't deal with that long words anyway. In my opinion, two lines label remains the best way to deal with it, but icons should be remain aligned in such a case.

userquin committed:
chore: add some vertical gap

userquin committed:
chore: include button preview translation

userquin commented:
> @userquin indeed, ellipsis is not advised. If that label can't be replaced by something shorter I'm not sure how to deal with this. French isn't as effective as english but we don't deal with that long words anyway. In my opinion, two lines label remains the best way to deal with it, but icons should be remain aligned in such a case.

We cannot do that, the bottom nav should have ~67px height, we can ask if the lang has some abbr. to use in that case otherwise there is no solution for this.

shuuji3 commented:
It seems that we need to find a short string for this new button label in some languages. For English, I think we can shorten "More menu" to "Menu" or something.

In various apps, developers themselves somehow come up with good short labels and if no localization is available, they fall back to English.

So how about adding new localization keys like `nav_short_label` next to `nav` keys here?
https://github.com/elk-zone/elk/blob/411fcc8/locales/en.json#L305

For example:

```diff
 "nav": {
     ....
     "conversations": "Conversations",
     ...
     "more_menu": "More menu",
     ...
+"nav_short_label": {
+    ....
+    "conversations": "Conversations",
+    ...
+    "more_menu": "Menu",
```

We can just copy the existing `nav` strings for languages that already have short labels and fall back to English while waiting for the update of other languages.

## Nav bars examples

I checked nav bars in various Android apps (in English, French, and German).

### Slack
![Screenshot_20250102-140357](https://github.com/user-attachments/assets/a40798f2-fd84-4818-a3a9-239aaa107600)
![Screenshot_20250102-140328](https://github.com/user-attachments/assets/ab40fb51-b0c8-4464-8f7d-c1ffb44989a2)
![Screenshot_20250102-140305](https://github.com/user-attachments/assets/3e9b8f8d-0d21-46a5-9487-3d474c9043dd)

### Play Store

![Screenshot_20250102-140728](https://github.com/user-attachments/assets/2cf75729-2428-42b7-bc06-ebc8da4f16fb)
![Screenshot_20250102-135909](https://github.com/user-attachments/assets/795bb60c-0d44-48d6-9054-28dac97a86dd)

### Contacts
![Screenshot_20250102-135645](https://github.com/user-attachments/assets/8060403f-1e41-4939-96dc-e09745cb3568)
![Screenshot_20250102-135641](https://github.com/user-attachments/assets/82cdf284-8436-4bdf-b2df-01b8156a5471)
![Screenshot_20250102-140653](https://github.com/user-attachments/assets/cab87184-e1ba-4242-bd58-4a33fc1fa9a3)

### Others
![Screenshot_20250102-135534](https://github.com/user-attachments/assets/9e53f555-9cc5-4f92-9976-c8b85ede73a6)
![Screenshot_20250102-140246](https://github.com/user-attachments/assets/a84eeb78-57d3-41ed-bc47-a51b5bf84c61)
![Screenshot_20250102-135756](https://github.com/user-attachments/assets/185fb60a-009a-4f9f-a213-1bab5c946cba)


### Google Fit and GitHub
They failed showing a good short label in German 😅 
![Screenshot_20250102-141111](https://github.com/user-attachments/assets/3e5de4ef-74ab-4069-979f-4a2bf7eef1df)
![Screenshot_20250102-141144](https://github.com/user-attachments/assets/54b5c1a2-adba-4ad9-afb9-72351f383e70)

shuuji3 commented:
It seems that we need to find a short string for this new button label in some languages. For English, I think we can shorten "More menu" to "More" or something.

In various apps, developers themselves somehow come up with good short labels and if no localization is available, they fall back to English.

So how about adding new localization keys like `nav_short_label` next to `nav` keys here?
https://github.com/elk-zone/elk/blob/411fcc8/locales/en.json#L305

For example:

```diff
 "nav": {
     ....
     "conversations": "Conversations",
     ...
     "more_menu": "More menu",
     ...
+"nav_short_label": {
+    ....
+    "conversations": "Conversations",
+    ...
+    "more_menu": "More",
```

We can just copy the existing `nav` strings for languages that already have short labels and fall back to English while waiting for the update of other languages. What do you think?

## Nav bars examples

I checked nav bars in various Android apps (in English, French, and German).

### Slack
![Screenshot_20250102-140357](https://github.com/user-attachments/assets/a40798f2-fd84-4818-a3a9-239aaa107600)
![Screenshot_20250102-140328](https://github.com/user-attachments/assets/ab40fb51-b0c8-4464-8f7d-c1ffb44989a2)
![Screenshot_20250102-140305](https://github.com/user-attachments/assets/3e9b8f8d-0d21-46a5-9487-3d474c9043dd)

### Play Store

![Screenshot_20250102-140728](https://github.com/user-attachments/assets/2cf75729-2428-42b7-bc06-ebc8da4f16fb)
![Screenshot_20250102-135909](https://github.com/user-attachments/assets/795bb60c-0d44-48d6-9054-28dac97a86dd)

### Contacts

![Screenshot_20250102-135641](https://github.com/user-attachments/assets/82cdf284-8436-4bdf-b2df-01b8156a5471)
![Screenshot_20250102-140653](https://github.com/user-attachments/assets/cab87184-e1ba-4242-bd58-4a33fc1fa9a3)

### Others
![Screenshot_20250102-135534](https://github.com/user-attachments/assets/9e53f555-9cc5-4f92-9976-c8b85ede73a6)
![Screenshot_20250102-140246](https://github.com/user-attachments/assets/a84eeb78-57d3-41ed-bc47-a51b5bf84c61)
![Screenshot_20250102-135756](https://github.com/user-attachments/assets/185fb60a-009a-4f9f-a213-1bab5c946cba)


### Google Fit and GitHub
They failed showing a good short label in German 😅 
![Screenshot_20250102-141111](https://github.com/user-attachments/assets/3e5de4ef-74ab-4069-979f-4a2bf7eef1df)
![399644292-54b5c1a2-adba-4ad9-afb9-72351f383e70](https://github.com/user-attachments/assets/82234e37-7860-4f1d-9e1d-8c387912b2d3)

shuuji3 committed:
style: adjust preview styling

shuuji3 commented:
^ A bit of style refinement (only for preview area)

![Screenshot of nav button settings page](https://github.com/user-attachments/assets/c79626eb-87f9-4f47-9893-dc2f6b0b4945)

![Screenshot of nav button settings page after adjusting spaces](https://github.com/user-attachments/assets/397e0ad3-fcc2-4778-bce0-8f82f0ddb695)

shuuji3 commented:
Showing ellipsis is suboptimal but maybe we need to use ellipsis for now to avoid overflow or two lines of label. For some languages having ellipsis labels, users already have choices either reducing to four buttons or hiding labels when they don't want to see ellipsis. Later, we can encourage translators to shorten those labels to avoid ellipsis as possible.

I'd like to keep the maximum number to five since it's better to avoid reducing functionality just for this label issue and the Material guide does not recommend more than 5 buttons[^1].

(Aside from the guideline, I don't know the current state but X app may not be reviewed by accessibility expert team so it may not be a good example in terms of accessibility[^2])

[^1]: [Navigation bar – Material Design 3](https://m3.material.io/components/navigation-bar/guidelines#8d570c9d-3822-4011-b974-1c7f00e02a5a)
[^2]: [Elon Musk just axed key Twitter teams like human rights, accessibility, AI ethics and curation | TechCrunch](https://techcrunch.com/2022/11/04/elon-musk-twitter-layoffs/)

shuuji3 committed:
style: try implementing nowrap and ellipsis

shuuji3 commented:
I'm trying to implement the ellipsis but the current CSS does not work as expected yet. But the two-line label issue was fixed.

![screenshot of nav bottom labels with narrow window width, labels are overflowing without ellipsis](https://github.com/user-attachments/assets/95369a4e-03e3-4a54-b528-c2764979b70b)

I referred this method: Truncate String With Ellipsis | CSS-Tricks - https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

shuuji3 committed:
Merge branch 'main' into shuuji3/a11y/show-nav-botton-label

shuuji3 committed:
fix: fix wrong label for "compose" nav button

ayo-run commented:
Hi @shuuji3 will you be able to resolve the conflicts? I want to check/test this next :) Thank you!

shuuji3 committed:
Merge remote-tracking branch 'upstream/main' into shuuji3/a11y/show-nav-botton-label

shuuji3 commented:
@ayo-run Just merged the latest main branch (there was 785 changed files)! The screenshot is the mobile viewport simulation for iPhone SE (375x667).

<img width="430" height="715" alt="screenshot of mobile view; no text overwrap" src="https://github.com/user-attachments/assets/638e9880-5010-4dd8-ad45-ed8f596f33e6" />

ayo-run reviewed (changes requested):
Tested on `Galaxy Z Fold 5` -- which is the thinnest screen / dimension on chrome dev tools -- and choosing navigation options with long labels. I think it is okay to completely hide the label if the screen is not wide enough, but would it be possible to hide all labels if this is the case? It looks inconsistent if only one is hidden. See screenshot:

<img width="433" height="391" alt="Screenshot from 2026-06-16 22-04-01" src="https://github.com/user-attachments/assets/bdfa22d1-5dc7-45c6-a3d2-b2e4abe1e07d" />
